ledger/ledger-api-domain/src/main/scala/com/digitalasset/ledger/api/domain.scala

```diff
@@ -415,4 +418,75 @@ object domain {
   object Feature {
     case object UserManagement extends Feature
   }
+
+  final case class JwksUrl(value: String) extends AnyVal {
+    def toURL = new URL(value)
+  }
+  object JwksUrl {
+    def fromString(value: String): Either[String, JwksUrl] =
+      Try(new URL(value)).toEither.left
+        .map { case NonFatal(e) =>
+          e.getMessage
+        }
+        .map(_ => JwksUrl(value))
+
+    def assertFromString(str: String): JwksUrl = fromString(str) match {
+      case Right(value) => value
+      case Left(err) => throw new IllegalArgumentException(err)
+    }
+  }
+
+  sealed trait IdentityProviderId {
+    def toRequestString: String
+
+    def toDb: Option[IdentityProviderId.Id]
+  }
+
+  object IdentityProviderId {
+    final case object Default extends IdentityProviderId {
+      override def toRequestString: String = ""
+      override def toDb: Option[Id] = None
+    }
+
+    final case class Id(value: Ref.LedgerString) extends IdentityProviderId {
+      override def toRequestString: String = value
+
+      override def toDb: Option[Id] = Some(this)
+    }
+
+    object Id {
```

Comment:
Do we need that object? Can we delete them and move its responsibilities to`IdentityProviderId`?

Reply:
There are currently 2 impl of the `IdentityProviderId` trait: `Default` and `Id` with concrete value. Not sure how to model `Default` then. `Option[IdentityProviderId]` was super confusing as could mean absence of the value (unknown) or Default case

Reply:
I'm guessing `Default` will never be persisted to the DB and its `def toDb` method will never be used?